drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/commands/ScenarioSimulationContext.java

```diff
@@ -156,28 +150,55 @@ public ScenarioGrid getBackgroundGrid() {
         return backgroundGridWidget.getScenarioGridPanel().getScenarioGrid();
     }
 
-    public ScenarioGridPanel getSelectedScenarioGridPanel() {
-        return getSelectedScenarioGridWidget().getScenarioGridPanel();
+    public Optional<ScenarioGridPanel> getSelectedScenarioGridPanel() {
+        return getSelectedScenarioGridWidget().map(ScenarioGridWidget::getScenarioGridPanel);
     }
 
-    public AbstractScesimGridModel getSelectedScenarioGridModel() {
-        return getSelectedScenarioGridLayer().getScenarioGrid().getModel();
+    public ScenarioGridPanel getScenarioGridPanelByGridWidget(GridWidget gridWidget) {
+        switch (gridWidget) {
+            case SIMULATION:
+                return simulationGridWidget.getScenarioGridPanel();
+            case BACKGROUND:
+                return backgroundGridWidget.getScenarioGridPanel();
+            default:
+                throw new IllegalArgumentException("Illegal GridWidget " + gridWidget);
+        }
     }
 
-    public ScenarioGridLayer getSelectedScenarioGridLayer() {
-        return getSelectedScenarioGridPanel().getScenarioGridLayer();
+    public AbstractScesimGridModel getAbstractScesimGridModelByGridWidget(GridWidget gridWidget) {
+        switch (gridWidget) {
+            case SIMULATION:
+                return simulationGridWidget.getModel();
+            case BACKGROUND:
+                return backgroundGridWidget.getModel();
+            default:
+                throw new IllegalArgumentException("Illegal GridWidget " + gridWidget);
+        }
     }
 
-    public Optional<GridWidget> getSelectedGRID_WIDGET() {
-        if (getSelectedScenarioGridModel() instanceof ScenarioGridModel) {
-            return Optional.of(GridWidget.SIMULATION);
-        } else if (getSelectedScenarioGridModel() instanceof BackgroundGridModel) {
-            return Optional.of(GridWidget.BACKGROUND);
-        } else {
-            return Optional.empty();
+    public <T extends AbstractScesimData> AbstractScesimModel<T> getAbstractScesimModelByGridWidget(GridWidget gridWidget) {
+        switch (gridWidget) {
+            case SIMULATION:
+                return (AbstractScesimModel<T>) status.getSimulation();
+            case BACKGROUND:
+                return (AbstractScesimModel<T>) status.getBackground();
+            default:
+                throw new IllegalArgumentException("Illegal GridWidget " + gridWidget);
```

Comment:
This should never happen so what about add a test like
> GridWidget.values().foreach(elem -> context.getAbstractScesimModelByGridWidget(elem));

so it will fail as soon as we will add a new GridWidget entry without update this method?